File: 0020-valid-parentheses/0020-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        if(s.length()==1){
            return false;
        }else{
            Stack<Character> data = new Stack<>();
            data.push('r');
            for(int i =0;i<s.length();i++){
                char temp = s.charAt(i);
                if(temp == '(' || temp=='[' || temp == '{'){
                    data.push(temp);
                }else if(temp == ')'){
                    if(data.peek()=='('){
                        data.pop();
                    }else{
                        return false;
                    }
                }else if(temp=='}'){
                    if(data.peek()=='{'){
                        data.pop();
                    }else{
                        return false;
                    }
                }else{
                    if(data.peek()=='['){
                        data.pop();
                    }else{
                        return false;
                    }
                }
            }
            if(data.peek()=='r'){
                return true;
            }else{
                return false;
            }


        }
        


    }
}

File: 0020-valid-parentheses/0020-valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {

        map<char,char> myMap;
        myMap['['] = ']';
        myMap['('] = ')';
        myMap['{'] = '}';

        stack<char> myStack;
        int length = s.size();
        for(int i = 0;i<length;i++){
            char curr = s.at(i);
            if(curr == '{' || curr == '(' || curr == '['){
                myStack.push(curr);
            }else{
                if(myStack.empty()){
                    return false;
                }
                char topChar = myStack.top();
                if(myMap[topChar] == curr){
                    myStack.pop();
                }else{
                    return false;
                }
            }
        }
        return myStack.empty();
    }
};